lib/foreman_maintain/concerns/system_helpers.rb

```diff
@@ -26,16 +26,66 @@ def build
         end
       end
 
-      def hostname
-        execute('hostname -f')
+      def check_min_version(name, minimal_version)
+        current_version = package_version(name)
+        if current_version
+          return current_version >= version(minimal_version)
+        end
+      end
+
+      def downstream_installation?
+        execute?('rpm -q satellite') ||
+          (execute('rpm -q foreman') =~ /sat.noarch/)
+      end
+
+      def execute?(command, input = nil)
+        execute(command, :stdin => input)
+        $CHILD_STATUS.success?
+      end
+
+      def execute!(command, options = {})
+        command_runner = Utils::CommandRunner.new(logger, command, options)
+        execution.puts '' if command_runner.interactive? && respond_to?(:execution)
+        command_runner.run
+        if command_runner.success?
+          command_runner.output
+        else
+          raise command_runner.execution_error
+        end
+      end
+
+      def execute(command, options = {})
+        command_runner = Utils::CommandRunner.new(logger, command, options)
+        execution.puts '' if command_runner.interactive? && respond_to?(:execution)
+        command_runner.run
+        command_runner.output
       end
 
       def file_exists?(filename)
         File.exist?(filename)
       end
 
-      def version(value)
-        Version.new(value)
+      def find_package(name)
+        execute(%(rpm -qa '#{name}'))
```

Comment:
the `qa` variant is too slow. `rpm -q ` should be used instead. Also make sure it returns nil/false if the pacakge is not found: currently it returns '' instead: therefore the feature looks like it's present even though it isn't

Reply:
Tests running in `< 6 seconds` now.
Thanks! @iNecas